pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -70,19 +70,25 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
     using GeneratorKernelMap =
         std::unordered_map<GeneratorOperation, KernelType>;
     using MatrixKernelMap = std::unordered_map<MatrixOperation, KernelType>;
+    using SparseMatrixKernelMap =
+        std::unordered_map<SparseMatrixOperation, KernelType>;
     using ControlledGateKernelMap =
         std::unordered_map<ControlledGateOperation, KernelType>;
     using ControlledGeneratorKernelMap =
         std::unordered_map<ControlledGeneratorOperation, KernelType>;
     using ControlledMatrixKernelMap =
         std::unordered_map<ControlledMatrixOperation, KernelType>;
+    using ControlledSparseMatrixKernelMap =
+        std::unordered_map<ControlledSparseMatrixOperation, KernelType>;
 
     GateKernelMap kernel_for_gates_;
     GeneratorKernelMap kernel_for_generators_;
     MatrixKernelMap kernel_for_matrices_;
+    SparseMatrixKernelMap kernel_for_sparse_matrices_;
```

Comment:
We could use `MatrixKernelMap` for both sparse and dense representations. :thinking:  Could you please elaborate on this decision? --- per our last discussions, we wanted to follow the SparseHamiltonian design practices for SparseQubitUnitary. I thought we don't need to go for new kernels maps and enums, instead we could re-use the existing ones and add `SparseMultiQubitOp` as a new operator.